### PR TITLE
refactor: do not cache response for preview branch and update service worker runtime cache pattern

### PR DESCRIPTION
**Pull request by nickhsine**
### Change Reason
Editors report that preview server (keystone-preview.twreporter.org) didn't show the latest content after modifying it in the keystone-editor.

After digging out, I guess the reason is the content is cached by browser.
Therefore, we remove `Cache-Control: public, max-age=300` response header for preview server.

### CODE REVIEW

**Comment by taylrj** [comment]
LGTM

File: src/express/server.js [hunk 1]
```diff
@@ -174,7 +174,10 @@ class ExpressServer {
         const statusCode = _.get(routerStaticContext, 'statusCode', statusOK)
         if (!res.headersSent && statusCode < statusRedirect) {
           const idToken = _.get(req, 'cookies.id_token')
-          if (idToken) {
+          if (
+            idToken ||
+            globalEnv.releaseBranch === releaseBranchConsts.preview
```
Comment (babygoat) [inline src/express/server.js:179]: The logic here is getting complicate. I suggest we should extract the logic here to a separate function like `IsCacheable()`. Remove the comment `//not to cache personal response` below. Leave the comments to indicate in what situation the cache does not apply.

**Comment by nickhsine** [comment]
Updated.

File: src/express/server.js [hunk 2]
```diff
@@ -160,32 +163,48 @@ class ExpressServer {
...
+        const statusCode = _.get(
+          routerStaticContext,
+          'statusCode',
+          statusCodeConst.ok
+        )
+        res.locals.statusCode = statusCode
+        next()
+      },
+      function setResponseHeader(req, res, next) {
+        if (globalEnv.releaseBranch === releaseBranchConsts.preview) {
```
Comment (babygoat) [inline src/express/server.js:186]: Leave the comment here to describe why we disable cache on preview branch

**Comment by babygoat** [comment]
Also, minor suggestion, split the statusCodeConst changes with the function extract patch for better review.

File: src/express/server.js (hunk 2 above)
Comment (nickhsine) [reply src/express/server.js:188]: Oops. Forget to add comments.

**Comment by nickhsine** [comment]
Updated.

**Comment by babygoat** [comment]
LGTM

**Comment by nickhsine** [comment]
Updated according to your suggestion.

**Comment by babygoat** [comment]
LGTM

**Pull request merged**